app/agents/AbstractAgent.ts

```diff
@@ -126,6 +129,13 @@ export abstract class AbstractAgent implements IAgent {
       throw this.err('Constructor not initialized');
     }
 
+    // setting data source
+    if (this.network.source === 'blockchain') {
+      this.source = new BlockchainSource(this.network, this.contract);
+    } else if (this.network.source === 'subgraph' && this.network.graphUrl) {
+      this.source = new SubgraphSource(this.network, this.contract);
+    }
```

Comment:
I think we need to throw an error if neither were configured

Reply:
yeah, you're right. Added at network module special condition - if source not set, than use blockchain by default:

```
if (networkConfig.source) {
  this.source = networkConfig.source;
} else {
  this.source = 'blockchain';
}
```